TypeCobol/Compiler/CodeElements/Entry/DataType.cs

```diff
@@ -23,18 +24,34 @@ public class DataType : ICobolLanguageLevel, IVisitable {
 
 		public override string ToString() { return Name; }
 
-		public override int GetHashCode() { return Name.GetHashCode(); }
 	    public bool AcceptASTVisitor(IASTVisitor astVisitor) {
 	        return astVisitor.Visit(this);
 	    }
 
 	    public override bool Equals(object obj) {
-			var other = obj as DataType;
-			if (other == null) return false;
-			return other == this;
-		}
+            return Equals(obj as DataType);
+        }
+
+        public bool Equals(DataType dataType)
+        {
+            if (Object.ReferenceEquals(this, dataType)) return true;
+            if (Object.ReferenceEquals(null, dataType)) return false;
+
+            return Name.Equals(dataType.Name, StringComparison.OrdinalIgnoreCase);
+        }
+
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                var hashCode = 13;
+                hashCode = (hashCode * 397) ^ Name.GetHashCode();
+
+                return hashCode;
+            }
+        }
```

Comment:
The original comparison mechanism used the CobolLanguageLevel because basically the idea was to use reference equality for Cobol 85 DataTypes and then compare names for other types.
I think we should use all three fields in Equals/GetHashCode implementations just to be on the safe side.

Reply:
done.
https://github.com/TypeCobolTeam/TypeCobol/pull/1600/commits/8bb22a0bed7c51638872bc38edfbcb12c9e2ab34